includes/Modules/Analytics_4/Settings.php

```diff
@@ -39,29 +39,24 @@ public function register() {
 		parent::register();
 
 		$this->register_owned_keys();
-	}
 
-	/**
-	 * Gets Analytics 4 settings.
-	 *
-	 * @since 1.99.0
-	 *
-	 * @return array Analytics 4 settings, or default if not set.
-	 */
-	public function get() {
-		$value = parent::get();
-
-		// This is a temporary solution to keep using the Analytics ownerID setting
-		// as the main source of truth for the Analytics 4 ownerID value.
-		//
-		// This is needed because currently the Analytics 4 functionality is separated
-		// from the Analytics module and we need to keep the ownerID synced between two
-		// modules. We will remove this hack when UA is sunset and only both modules
-		// are merged.
-		$analytics_settings = ( new Analytics_Settings( $this->options ) )->get();
-		$value['ownerID']   = $analytics_settings['ownerID'];
-
-		return $value;
+		// Since migration of Analytics module settings into Analytics 4 settings,
+		// if data was saved previously, it needs to be recovered.
+		add_filter(
+			'option_' . self::OPTION,
+			function ( $option ) {
+				if ( is_array( $option ) ) {
+					$missing_settings = $this->retrieve_missing_analytics_4_settings( $option );
+
+					if ( ! empty( $missing_settings ) ) {
+						return $option + $missing_settings;
+					}
+				}
+
+				return $option;
+			},
```

Comment:
I think we can move this logic into the `retrieve_missing_...` method and use `add_filter( 'option_' . self::OPTION, array( $this, 'retrieve_missing_...' ) )`.